Dockerfile.test.cpu

```diff
@@ -236,7 +236,7 @@ RUN if [[ ${MPI_KIND} == "ONECCL" ]]; then \
     fi; \
     cd /horovod && \
     python setup.py sdist && \
-    bash -c "${HOROVOD_BUILD_FLAGS} HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITH_PYTORCH=1 HOROVOD_WITH_MXNET=1 pip install --no-cache-dir -v $(ls /horovod/dist/horovod-*.tar.gz)[spark,ray]"
```

Comment:
This is not a breaking change, as Horovod can still be installed via the old HOROVOD_WITH_*=1 vars using `--no-build-isolation`, right?

    HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITH_PYTORCH=1 HOROVOD_WITH_MXNET=1 pip install --no-build-isolation ...

Can we somehow imply the `--no-build-isolation` when those HOROVOD_WITH_* vars are `1`? Otherwise we may be considered a breaking change...